pkg/querysharding/analyzer.go

```diff
@@ -20,12 +25,41 @@ var nonShardableFuncs = []string{
 
 // NewQueryAnalyzer creates a new QueryAnalyzer.
 func NewQueryAnalyzer() *QueryAnalyzer {
-	return &QueryAnalyzer{}
+	cache, err := lru.New(256)
+	logger := logging.NewLogger("warn", "", "")
+
+	if err != nil {
+		logger.Log("Error Creating LRU Cache: ", err)
```

Comment:
We should return error instead. No need to log since the error we returned will be logged I believe.
If that's the only place we need to log then we can probably remove the logger.

Reply:
Ok